Classes/Utility/FLEXRuntimeUtility.m

```diff
@@ -676,21 +689,63 @@ + (NSString *)readableTypeForEncoding:(NSString *)encodingString
     // See https://github.com/nygard/class-dump/blob/master/Source/CDType.m
     // Warning: this method uses multiple middle returns and macros to cut down on boilerplate.
     // The use of macros here was inspired by https://www.mikeash.com/pyblog/friday-qa-2013-02-08-lets-build-key-value-coding.html
-    const char *encodingCString = encodingString.UTF8String;
+    const char *encodingCString = [encodingString UTF8String];
+
+    // Some fields have a name, such as {Size=\"width\"d\"height\"d}, we need to extract the name out and recursive
+    const NSUInteger fieldNameOffset = [FLEXRuntimeUtility fieldNameOffsetForTypeEncoding:encodingCString];
+    if (fieldNameOffset > 0) {
+        // According to https://github.com/nygard/class-dump/commit/33fb5ed221810685f57c192e1ce8ab6054949a7c,
+        // there are some consecutive quoted strings, so use `_` to concatenate the names.
+        NSString *const fieldNamesString = [encodingString substringWithRange:NSMakeRange(0, fieldNameOffset)];
+        NSArray<NSString *> *const fieldNames = [fieldNamesString componentsSeparatedByString:[NSString stringWithFormat:@"%c", FLEXTypeEncodingQuote]];
+        NSMutableString *finalFieldNamesString = [NSMutableString string];
+        for (NSString *const fieldName in fieldNames) {
+            if (fieldName.length > 0) {
+                if (finalFieldNamesString.length > 0) {
+                    [finalFieldNamesString appendString:@"_"];
+                }
+                [finalFieldNamesString appendString:fieldName];
+            }
+        }
+        NSString *const recursiveType = [self readableTypeForEncoding:[encodingString substringFromIndex:fieldNameOffset]];
+        return [NSString stringWithFormat:@"%@ %@", recursiveType, finalFieldNamesString];
+    }
 
     // Objects
-    if (encodingCString[0] == FLEXTypeEncodingObjcObject) {
+    if (encodingCString[0] == '@') {
         NSString *class = [encodingString substringFromIndex:1];
         class = [class stringByReplacingOccurrencesOfString:@"\"" withString:@""];
-        if (class.length == 0 || (class.length == 1 && [class characterAtIndex:0] == FLEXTypeEncodingUnknown)) {
+        if ([class length] == 0 || ([class length] == 1 && [class characterAtIndex:0] == FLEXTypeEncodingUnknown)) {
```

Comment:
Same as first, why the regression?

Reply:
I forgot why, probably because I’m working on a forked repo first and copied over some changes. I will revert all the syntax changes.